dev-qt/qt-creator/qt-creator-8.0.0.ebuild

```diff
@@ -0,0 +1,311 @@
+
+inherit cmake llvm optfeature virtualx xdg
+
+DESCRIPTION="Lightweight IDE for C++/QML development centering around Qt"
+HOMEPAGE="https://doc.qt.io/qtcreator/"
+
+if [[ ${PV} == *9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://code.qt.io/${PN}/${PN}.git"
+	EGIT_SUBMODULES=(
+		perfparser
+		qtscript # Need the dev branch
+		src/libs/qlitehtml
+		src/libs/qlitehtml/src/3rdparty/litehtml
+	)
+	# PV=7.9999 -> EGIT_BRANCH=7.0
+	[[ ${PV} =~ [0-9]+\.9999 ]] && EGIT_BRANCH=$(ver_cut 1).0
+	# PV=7.2.9999 -> EGIT_BRANCH=7.2
+	[[ ${PV} =~ [0-9]+\.[0-9]+\.9999 ]] && EGIT_BRANCH=${PV%.9999}
+else
+	MY_PV=${PV/_/-}
+	MY_P=${PN}-opensource-src-${MY_PV}
+	[[ ${MY_PV} == ${PV} ]] && MY_REL=official || MY_REL=development
+	SRC_URI="https://download.qt.io/${MY_REL}_releases/${PN/-}/$(ver_cut 1-2)/${MY_PV}/${MY_P}.tar.xz"
+	S=${WORKDIR}/${MY_P}
+	KEYWORDS="~amd64 ~arm ~x86"
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+
+IUSE="+clang debug designer doc imageviewer serialterminal systemd test +qml qt6 wayland webengine"
+
+# TODO: Sort out the web of test deps
+RESTRICT="test"
+
+# minimum Qt version required
+QT5_PV="5.15.2:5="
+QT6_PV="6.2.0:6"
+
+BDEPEND="
+	qt6? (
+		>=dev-qt/qttools-${QT6_PV}[linguist(+)]
+		doc? ( >=dev-qt/qttools-${QT6_PV}[qdoc(+)] )
+	)
+	!qt6? (
+		>=dev-qt/linguist-tools-${QT5_PV}
+		doc? ( >=dev-qt/qdoc-${QT5_PV} )
+	)
+"
+CDEPEND="
+	app-arch/zstd
+	dev-libs/elfutils
+	<sys-devel/clang-$((LLVM_MAX_SLOT + 1)):=
+	|| (
+		sys-devel/clang:14
+		sys-devel/clang:13
+	)
+
+	clang? (
+		>=dev-cpp/yaml-cpp-0.6.2:=
+		sys-devel/clang:14=
+	)
+	qt6? (
+		>=dev-qt/qt5compat-${QT6_PV}
+		>=dev-qt/qtbase-${QT6_PV}[concurrent,gui,network,sql,widgets]
+		>=dev-qt/qtdeclarative-${QT6_PV}
+		>=dev-qt/qttools-${QT6_PV}[assistant(+)]
+
+		designer? ( >=dev-qt/qttools-${QT6_PV}[designer(+)] )
+		imageviewer? ( >=dev-qt/qtsvg-${QT6_PV} )
+		qml? (
+			>=dev-qt/qtquick3d-${QT6_PV}
+			>=dev-qt/qtshadertools-${QT6_PV}
+		)
+		serialterminal? ( >=dev-qt/qtserialport-${QT6_PV} )
+		webengine? ( >=dev-qt/qtwebengine-${QT6_PV}[widgets] )
+	)
+	!qt6? (
+		>=dev-qt/qtconcurrent-${QT5_PV}
+		>=dev-qt/qtcore-${QT5_PV}
+		>=dev-qt/qtdeclarative-${QT5_PV}[widgets]
+		>=dev-qt/qthelp-${QT5_PV}
+		>=dev-qt/qtgui-${QT5_PV}
+		>=dev-qt/qtnetwork-${QT5_PV}[ssl]
+		>=dev-qt/qtprintsupport-${QT5_PV}
+		>=dev-qt/qtquickcontrols-${QT5_PV}
+		>=dev-qt/qtsql-${QT5_PV}[sqlite]
+		>=dev-qt/qtwidgets-${QT5_PV}
+		>=dev-qt/qtxml-${QT5_PV}
+		>=kde-frameworks/syntax-highlighting-5.87:5
+
+		designer? ( >=dev-qt/designer-${QT5_PV} )
+		imageviewer? ( >=dev-qt/qtsvg-${QT5_PV} )
+		serialterminal? ( >=dev-qt/qtserialport-${QT5_PV} )
+		webengine? ( >=dev-qt/qtwebengine-${QT5_PV}[widgets] )
+	)
+	systemd? ( sys-apps/systemd:= )
+	!webengine? ( dev-libs/gumbo )
+"
+DEPEND="
+	${CDEPEND}
+	test? (
+		dev-cpp/benchmark
+		dev-cpp/eigen
+		dev-libs/boost
+
+		qt6? ( >=dev-qt/qtbase-${QT6_PV}[test] )
+		!qt6? ( >=dev-qt/qttest-${QT5_PV} )
+	)
+"
+RDEPEND="
+	${CDEPEND}
+	wayland? (
```

Comment:
qtcreator has wayland support since 7.0.0.  This enforces a built-with-use dependence.  It's a strict RDEPEND because qtcreator will build fine without `dev-qt/qtbase-${QT6_PV}[egl]` and `dev-qt/qtwayland-${QT6_PV}` but as seen in https://github.com/gentoo/qt/pull/252#issuecomment-1165732612  opening qtcreator in a wayland environment leads to:
```
qt.qpa.wayland: Failed to load client buffer integration: "wayland-egl"
qt.qpa.wayland: Available client buffer integrations: QList()
```

This issue is resolved as soon as those deps are emerged.  The QT5 dep, i.e., `!qt6? ( >=dev-qt/qtgui-${QT5_PV}[wayland] )`, may be unnecessary as there's no KDE6 wayland environment to test whether, if built against QT5 but without `>=dev-qt/qtgui-${QT5_PV}[wayland]`, it would display a similar error.  Either way it acts as a sanity check, if nothing else.

Reply:
wrt wayland, there's https://bugs.gentoo.org/864509 given will probably need a central way to pull qtwayland as having this on every packages will be kind of silly and qtwayland should depend on whatever it needs, don't think it needs to be handled in qtcreator

Reply:
>and qtwayland should depend on whatever it needs

Albeit not sure if EGL is always needed, it depends on opengl right now which maybe is lacking?